sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Helpers/SymbolicExecution/InvocationVisitor.cs

```diff
@@ -79,11 +79,14 @@ internal ProgramState ProcessInvocation()
 
         private ProgramState HandleNameofExpression()
         {
-            SymbolicValue arg1;
-
-            var newProgramState = programState
-                .PopValue(out arg1)
-                .PopValue();
+            // First pop should handle nameof argument but if nameof is empty (invalid) there is only one item in the
+            // ExpressionStack (the nameof itself) so we don't want to pop the second time.
+            // NB: We decided not to stop the execution of the DataFlow Analysis.
+            var newProgramState = programState.PopValue();
+            if (!newProgramState.ExpressionStack.IsEmpty)

```

Comment:
Why not check instead that the invocation has an argument or not? The invocation object is available in the calling method.